pkg/strategy/xdepthmaker/profitfixer.go

```diff
@@ -0,0 +1,84 @@
+package xdepthmaker
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"golang.org/x/sync/errgroup"
+
+	"github.com/c9s/bbgo/pkg/exchange/batch"
+	"github.com/c9s/bbgo/pkg/types"
+)
+
+type ProfitFixerConfig struct {
+	TradesSince types.Time `json:"tradesSince,omitempty"`
+}
+
+// ProfitFixer implements a trade history based profit fixer
+type ProfitFixer struct {
+	market types.Market
+
+	sessions map[string]types.ExchangeTradeHistoryService
+}
+
+func NewProfitFixer(market types.Market) *ProfitFixer {
+	return &ProfitFixer{
+		market:   market,
+		sessions: make(map[string]types.ExchangeTradeHistoryService),
+	}
+}
+
+func (f *ProfitFixer) AddExchange(sessionName string, service types.ExchangeTradeHistoryService) {
+	f.sessions[sessionName] = service
+}
+
+func (f *ProfitFixer) batchQueryTrades(
+	ctx context.Context,
+	service types.ExchangeTradeHistoryService,
+	symbol string,
+	since time.Time,
```

Comment:
Will it be more flexible if we can specify the `since`, too?

Reply:
do you mean `until` ?

Reply:
added `until`